src/ray/core_worker/object_recovery_manager.cc

```diff
@@ -51,6 +52,13 @@ bool ObjectRecoveryManager::RecoverObject(const ObjectID &object_id) {
       // duplicate restarts of the same object.
       already_pending_recovery = !objects_pending_recovery_.insert(object_id).second;
     }
+    if (delete_from_memory_store_if_recovering) {
+      // Delete the objects from the in-memory store to indicate that they are not
+      // available. The object recovery manager will guarantee that a new value
+      // will eventually be stored for the objects (either an
+      // UnreconstructableError or a value reconstructed from lineage).
+      in_memory_store_->Delete({object_id});
+    }
```

Comment:
Specially, line 81 is the case where we don't want to delete the object from memory store since the object doesn't need recovery.